### PR TITLE
Update PR template to select either data plane or control plane template

### PR DESCRIPTION
**Pull request by mikekistler**
This PR updates the base PR template to let the user choose a template for a data plane API or control plane (ARM) API.

### CODE REVIEW

**Review by weshaggard — COMMENTED** [review]
The change is expected but I will defer to @akning-ms to make sure that any automation we have is using the new templates by name now.

**Review by akning-ms — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (akning-ms) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Looks Link is wrong, I tryied it, it reports 404 eror

Comment (mikekistler) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: The link is relative so it only works correctly when you start from the expected place, which is within the Pull Request dialog in GitHub. When you request to open a PR in this repo you will be put into a dialog where the URL looks like:

https://github.com/Azure/azure-rest-api-specs/compare/main...<branch>

This is where the PR template will be shown. Clicking the link when on that page will give you a URL that looks like

https://github.com/Azure/azure-rest-api-specs/compare/main...<branch>?expand=1&template=data_plane_template.md

and that link will resolve correctly.  I just tested it again.

**Comment by mikekistler** [comment]
I updated the PR to resolve a conflict with [this change](https://github.com/Azure/azure-rest-api-specs/commit/f699f9210bfb53e5841bd45c5637a122a17ad32b) and then force-pushed.

I made the corresponding change to .github/PULL_REQUEST_TEMPLATE/control_plane_template.md.

File: .github/PULL_REQUEST_TEMPLATE/control_plane_template.md
Comment (weshaggard) [inline .github/PULL_REQUEST_TEMPLATE/control_plane_template.md:36]:
```suggestion
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
```

**Comment by raych1** [comment]
@akning-ms , please hold off the merge as I need to switch the template location used in the shift left pipeline.

**Pull request merged**
